pkg/virt-controller/watch/vm_test.go

```diff
@@ -685,9 +693,7 @@ var _ = Describe("VirtualMachine", func() {
 				createCount := 0
 				shouldExpectDataVolumeCreation(vm.UID, map[string]string{"kubevirt.io/created-by": ""}, map[string]string{}, &createCount)
 
-				if fail {
-					vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Return(vm, nil)
-				}
+				vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Return(vm, nil)
```

Comment:
maybe we should be setting the vm.Status.PrintableStatus to the expected value here to avoid the update always occurring. This changes the meaning of the test a bit by always having the UpdateStatus hit.

I think this goes for all the existing tests where `vmInterface.EXPECT().UpdateStatus(gomock.Any()).Times(1).Return(vm, nil)` has been added to satisfy the PrintableStatus being set during execute()

Not having an UpdateStatus() client call invoked is part of these test cases. We want to make sure that UpdateStatus is only called when something expected changes. By having it always be expected due to the PrintableStatus change, then we could accidently leak in other unexpected changes as well that would invoke UpdateStatus.

does that make sense?

Reply:
Yes, that makes perfect sense. I'll modify accordingly.

Reply:
btw, fixing this made me realize that the current implementation doesn't detect when a VM with pending dataVolumeTemplates is being started (i.e., it stays in Stopped status up until the DVs are ready and the VMI gets created, and only then moves to Provisioning). I'll work on fixing that too.